### DISCUSSION TITLE
Can't delete interface

### DISCUSSION THREAD

**Discussion by gbrownie** [netbox-community/netbox#15755] [Help Wanted!] [answered]
While trying to delete some interfaces on a device, I received the error below. I am not getting this error with other interfaces. 

```
<class 'django.db.utils.IntegrityError'>

update or delete on table "dcim_interface" violates foreign key constraint "netdoc_arptableentry_interface_id_6b216710_fk_dcim_interface_id" on table "netdoc_arptableentry"
DETAIL:  Key (id)=(9294) is still referenced from table "netdoc_arptableentry".

Python version: 3.10.12
NetBox version: 3.7.4
Plugins: 
  netbox_documents: 0.6.3
  netbox_topology_views: 3.9.0
  phonebox_plugin: v0.0.7-beta.2
```

**Comment by a084ed22** [comment, marked as answer]
by the looks of it you used to use the netdoc plugin. Despite not being installed/active anymore, the database structures for it are still in place, and you have pending references to the interface you're trying to delete.

If you're not using netdoc anymore and don't care about the data left over, try to drop the netdoc_* tables.

**Reply by gbrownie** [reply]
Thank you! I still need to do some more testing and I'm pretty new to all of this but this is what I did. I dropped all the netdoc tables with 'drop table netdoc_* cascade;' I then tried removing the netdoc plugin from 'local_requirements.txt' and run upgrade.sh. It failed stating I needed the netdoc plugin. So, I added the netdoc plugin back and re-ran upgrade.sh. I check netdoc and the couple of files I had in there were still there? I then tried to delete the problematic device and it did delete. A bit confused how dropping the tables didn't get rid of the documents though.

**Reply by a084ed22** [reply]
very likely you still are attempting to load the plugin in netbox/netbox/configuration.py. Be sure that it's not listed in PLUGINS or PLUGINS_CONFIG.

**Reply by gbrownie** [reply]
Right, I did forget to remove it from configuration.py. Ok, its completely removed now and I have removed the problematic device. Thank you again.
